rsocket-core/src/test/java/io/rsocket/loadbalance/LoadbalanceTest.java

```diff
@@ -51,12 +57,14 @@ public void shouldDeliverAllTheRequestsWithRoundRobinStrategy() {
       final TestPublisher<List<LoadbalanceTarget>> source = TestPublisher.create();
       final RSocketPool rSocketPool =
           new RSocketPool(rSocketConnectorMock, source, new RoundRobinLoadbalanceStrategy());
+      final CountDownLatch requestLatch = new CountDownLatch(1000);
 
       RaceTestUtils.race(
           () -> {
             for (int j = 0; j < 1000; j++) {
               Mono.defer(() -> rSocketPool.select().fireAndForget(EmptyPayload.INSTANCE))
                   .retry()
+                  .doFinally(s -> requestLatch.countDown())
```

Comment:
retry() is a sync op, thus all execution should be within `RaceTestUtils.race` and once the racing is done, nothing should be working anymore, hence, I suspect, the bug is in a different place ( can we try 100k cycles on CI to ensure it is not reproing even with CDL (potentially CDL slowdowns racing, hence tests are passing)

Reply:
I can remove this part. I think the bug is as I explained in https://github.com/rsocket/rsocket-java/pull/983#issuecomment-781310234. That is a clear issue.